native/android/src/main/java/com/nozbe/watermelondb/DatabaseDriver.kt

```diff
@@ -163,12 +163,19 @@ class DatabaseDriver(context: Context, dbName: String) {
 
     fun close() = database.close()
 
-    private fun markAsCached(id: RecordID) {
+    private fun markAsCached(table: TableName, id: RecordID) {
         log?.info("Mark as cached $id")
-        cachedRecords.add(id)
+        var cache = cachedRecords[table]
+        if (cache == null) {
+            cache = mutableListOf<RecordID>()
+            cachedRecords[table] = cache!!
```

Comment:
`mutableListOf<RecordID>()` is creating non-nullable type of object so double bang is unnecessary